modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BatchEngineExportTaskExecutorTest.java

```diff
@@ -254,6 +259,88 @@ public void testExportBlogPostingsToXLSFileWithFieldNames()
 			true);
 	}
 
+	@Test
+	public void testExportBlogPostingsWithoutPersistingContent()
+		throws Exception {
+
+		List<BlogsEntry> blogsEntries = addBlogsEntries();
+
+		_batchEngineExportTask =
+			_batchEngineExportTaskLocalService.addBatchEngineExportTask(
+				null, user.getCompanyId(), user.getUserId(), null,
+				BlogPosting.class.getName(), "JSON",
+				BatchEngineTaskExecuteStatus.INITIAL.name(), null, _parameters,
+				null);
+
+		BatchEngineExportTaskExecutor.Result result =
+			_batchEngineExportTaskExecutor.execute(
+				_batchEngineExportTask,
+				new BatchEngineExportTaskExecutor.Settings() {
+
+					@Override
+					public boolean isCompressContent() {
+						return false;
+					}
+
+					@Override
+					public boolean isPersistContent() {
+						return false;
+					}
+
+				});
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray(
+			StringUtil.read(result.getInputStream()));
+
+		Assert.assertTrue(jsonArray.length() >= blogsEntries.size());
+
+		_batchEngineExportTask =
+			_batchEngineExportTaskLocalService.getBatchEngineExportTask(
+				_batchEngineExportTask.getBatchEngineExportTaskId());
+
+		Assert.assertEquals(
+			_batchEngineExportTask, result.getBatchEngineExportTask());
+
+		Assert.assertEquals(
+			BatchEngineTaskExecuteStatus.COMPLETED.toString(),
+			_batchEngineExportTask.getExecuteStatus());
+		Assert.assertEquals(
+			blogsEntries.size(),
+			_batchEngineExportTask.getProcessedItemsCount());
+		Assert.assertEquals(
+			blogsEntries.size(), _batchEngineExportTask.getTotalItemsCount());
+
+		Blob content = _batchEngineExportTask.getContent();
```

Comment:
Do you know which is the reason why the Blob contains an empty String instead of just a null value?

https://github.com/liferay/liferay-portal/blob/02e753d35ffc6e8c702e66f52f3f47786557f757/modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/service/impl/BatchEngineExportTaskLocalServiceImpl.java#L53

Reply:
I don't know, we have not investigated that. It's the current behaviour.